pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -248,3 +257,69 @@ def insert_argument(self):
         Get the argument which is passed to insert().
         """
         return self._insert_arg
+
+#==============================================================================
+class ListExtend(PyccelInternalFunction):
+    """
+    Represents a call to the .extend() method.
+
+    Represents a call to the .extend() method of an object with a list type,
+    which adds items of an iterable (list, tuple, dictionary, etc) at the end
+    of a list.
+    This method returns `None`.
+    The extend method is called as follows:
+
+    >>> a = [1]
+    >>> a.extend([2])
+    >>> print(a)
+    [1, 2]
+
+    Parameters
+    ----------
+    list_variable : TypedAstNode
+        The list object which the method is called from.
+    
+    new_elem : TypedAstNode
+        The argument passed to extend() method.
+    """
+    __slots__ = ("_list_variable", "_extend_arg")
+    _attribute_nodes = ("_list_variable", "_extend_arg")
+    _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = None
+    _class_type = NativeVoid()
+    name = 'extend'
+
+    def __init__(self, list_variable, new_elem) -> None:
```

Comment:
updated in 4190b7632976e974227dfee19adad833d3b7de5e
`__init__()` method is deprecated since there's now `_visit_ListExtend` implementation that handles the call to `extend()`